test/e2e/utils.go

```diff
@@ -0,0 +1,231 @@
+// +build e2e
+
+package e2e
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// lookForStringInPodExec looks for expectedString in the output of command
+// executed in the specified pod container.
+func lookForStringInPodExec(ns, pod, container string, command []string, expectedString string, timeout time.Duration) error {
+	cmd, err := lookupCmdPath("oc")
+	if err != nil {
+		return err
+	}
+	args := []string{"exec", pod, "-c", container, fmt.Sprintf("--namespace=%v", ns), "--"}
+	args = append(args, command...)
+	if err := lookForString(cmd, args, expectedString, timeout); err != nil {
+		return err
+	}
+	return nil
+}
+
+// lookForStringInPodLog looks for the given string in the log of a specific pod container
+func lookForStringInPodLog(ns, pod, container, expectedString string, timeout time.Duration) error {
+	cmd, err := lookupCmdPath("oc")
+	if err != nil {
+		return err
+	}
+	args := []string{"logs", pod, "-c", container, fmt.Sprintf("--namespace=%v", ns)}
+	if err := lookForString(cmd, args, expectedString, timeout); err != nil {
+		return err
+	}
+	return nil
+}
+
+// lookupCmdPath looks-up the path of cmd returning the full command
+// path or an error.
+func lookupCmdPath(cmdPath string) (string, error) {
+	cmdPath, err := exec.LookPath(cmdPath)
+	if err != nil {
+		return "", fmt.Errorf("failed to find the %s executable in the directories named by the PATH environment variable.", cmdPath)
+	}
+	return cmdPath, nil
```

Comment:
Is `lookupCmdPath` needed?  Looks like callers could call `exec.LookPath` directly.  If the executable is not found, `exec.LookPath` will return a reasonable error string itself: `exec: "oc": executable file not found in $PATH`.

Reply:
ack